04_numeronym/CameronSchafer/main_test.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// main() test
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+	expected := "[a11y K8s abc]\n"
+	actual := buf.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
+	}
+}
+
+// numeronyms()
+func TestNumeronyms(t *testing.T) {
+	// test cases with descriptions.
+	testCases := []struct {
+		description string
+		input       []string
+		expectedArr []string
+	}{
+		{description: "numeronyms('accessibility', 'Kubernetes', 'abc')",
+			input:       []string{"accessibility", "Kubernetes", "abc"},
+			expectedArr: []string{"a11y", "K8s", "abc"},
+		},
+		{description: "numeronyms('a', 's', 'd')", input: []string{"a", "s", "d"},
+			expectedArr: []string{"a", "s", "d"},
+		},
+		{description: "numeronyms('', '', '')", input: []string{"", "", ""},
+			expectedArr: []string{"", "", ""},
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		// t.Run creates a sub test and runs it like a normal test
+		t.Run(test.description, func(t *testing.T) {
+			resultArr := numeronyms(test.input...)
+			assert.Equalf(t, resultArr, test.expectedArr, "Unexpected output in %v\nexpected: %v,\nactual: %v",
```

Comment:
The testify library would give you the comparisons of expected vs actual, so not necessary to print "Unexpected output...."